daft/table/table.py

```diff
@@ -114,14 +114,39 @@ def concat(cls, to_merge: list[Table]) -> Table:
     ###
 
     def to_arrow(self) -> pa.Table:
-        python_fields = [field for field in self.schema() if field.dtype == DataType.python()]
+        python_fields = {field.name for field in self.schema() if field.dtype == DataType.python()}
         if python_fields:
-            raise ValueError(f"Cannot convert table to Arrow due to presence of Python columns f{python_fields}")
-        return pa.Table.from_batches([self._table.to_arrow_record_batch()])
+            table = {}
+            for colname in self.column_names():
+                column_series = self.get_column(colname)
+                if colname in python_fields:
+                    # TODO(Clark): Get the column as a top-level ndarray to ensure it remains contiguous.
+                    column = column_series.to_pylist()
+                    # TODO(Clark): Infer the tensor extension type even when the column is empty.
+                    # This will probably require encoding more information in the Daft type that we use to
+                    # represent tensors.
+                    if len(column) > 0 and isinstance(column[0], np.ndarray):
```

Comment:
@jaychia @samster25 If given an empty tensor column (i.e. an empty column that at one point contained tensors), this will currently read it as the `pa.null()` type rather than a tensor type.

Reply:
Seems fine to me, the underlying Table's schema isn't actually accessed by a user since they will only see the DataFrame-level schema